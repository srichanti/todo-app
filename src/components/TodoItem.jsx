import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const StyledList = styled.li`
  list-style: none;
  overflow: hidden;
  width: 100%;
  margin-bottom: 10px;
`

const StyledLabel = styled.label`
  float: left;
  cursor: pointer;
`

const StyledButton = styled.button`
  float: right;
  background: palevioletred;
  color: #fff;
  border-radius: 3px;
  border: 2px solid palevioletred;
  padding: 3px 10px;
  outline: none;
  cursor: pointer;
`

const TodoItem = props => {
  const { id, title } = props
  return (
    <StyledList>
      <StyledLabel htmlFor={id}>
        <input type="checkbox" id={id} /> {title}
      </StyledLabel>
      <StyledButton type="button">Delete</StyledButton>
    </StyledList>
  )
}
TodoItem.propTypes = {
  title: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired
}
export default TodoItem
