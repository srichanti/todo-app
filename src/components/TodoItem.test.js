import React from 'react'
import { shallow, mount } from 'enzyme'
import TodoItem from './TodoItem'

describe('TodoItem Component', () => {
  let shallowComponent
  let mountComponent
  let id
  let title

  beforeEach(() => {
    id = 2
    title = 'Unit test'
    shallowComponent = shallow(<TodoItem id={id} title={title} />)
    mountComponent = mount(<TodoItem id={id} title={title} />)
  })

  it('renders correctly', () => {
    expect(shallowComponent).toMatchSnapshot()
  })

  it('TodoItem requires id prop', () => {
    expect(mountComponent.props().id).toBeDefined()
  })
  it('TodoItem requires title prop', () => {
    expect(mountComponent.props().title).toBeDefined()
  })
  it('TodoItem renders Input', () => {
    expect(shallowComponent.find('input').length).toEqual(1)
  })
})
