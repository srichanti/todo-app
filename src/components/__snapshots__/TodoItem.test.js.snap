// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem Component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TodoItem
    id="todo-1"
    title="Do house chores"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <styled.label
          htmlFor="todo-1"
        >
          <input
            id="todo-1"
            type="checkbox"
          />
           
          Do house chores
        </styled.label>,
        <styled.button
          type="button"
        >
          Delete
        </styled.button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <input
              id="todo-1"
              type="checkbox"
            />,
            " ",
            "Do house chores",
          ],
          "htmlFor": "todo-1",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "todo-1",
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          " ",
          "Do house chores",
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": "Delete",
          "type": "button",
        },
        "ref": null,
        "rendered": "Delete",
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.label
            htmlFor="todo-1"
          >
            <input
              id="todo-1"
              type="checkbox"
            />
             
            Do house chores
          </styled.label>,
          <styled.button
            type="button"
          >
            Delete
          </styled.button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <input
                id="todo-1"
                type="checkbox"
              />,
              " ",
              "Do house chores",
            ],
            "htmlFor": "todo-1",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "todo-1",
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            " ",
            "Do house chores",
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "Delete",
            "type": "button",
          },
          "ref": null,
          "rendered": "Delete",
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
